src/actions/arcActions.ts

```diff
@@ -820,8 +820,8 @@ export function onProposalExecuted(avatarAddress: string, proposalId: string, ex
       proposal.reputationWhenExecuted = reputationWhenExecuted;
       proposal.winningVote = decision;
 
-      const gpProposalDetails = await votingMachineInstance.getProposal(proposalId);
-      proposal.proposer = gpProposalDetails.proposer; // Have to do this because redeem sets proposer to 0, to prevent future redemptions for proposer
+      // Have to do this because redeem sets proposer to 0, to prevent future redemptions for proposer
+      proposal.proposer = (await votingMachineInstance.NewProposal({_proposalId: proposalId}, {fromBlock: 0, toBlock: 'latest'}).get(undefined, -1))[0].args._proposer;
```

Comment:
hmm, wait, actually this is tricky. We actually want to know that the proposer is 0 here, so we don't show redemptions for the proposer. Or wait, i guess that was fixed on the Arc side? Did you test this?

Reply:
alchemy needs the `proposal.proposer` to always be non-zero which is guarenteed to be the case when fetching from the event, `getRedemptions(..)` gets the proposer from the blockchain so it'll be zero after a redeem. I don't understand the issue.

Reply:
yeah i hear that, but then seems like you can remove a few places where we check for whether proposer is 0x000...